### PR TITLE
Forcibly remove namespace

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10265)

### CODE REVIEW

**Comment by nebril** [comment]
test-gke K8sPolicyTest Basic Test Validate CNP update.*

**Comment by nebril** [comment]
test-gke K8sPolicy.*

**Comment by nebril** [comment]
test-gke K8sPolicy.*

**Comment by nebril** [comment]
test-gke K8sPolicy.*

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-gke K8sPolicy.*

**Comment by tgraf** [comment]
I would sleep better if we also remove all namespaces in `select-cluster.sh` to guarantee that we never drag the inconsistent state into new builds.

**Comment by nebril** [comment]
test-gke K8sPolicy.*

File: test/gke/select-cluster.sh [hunk 1]
```diff
@@ -31,6 +31,9 @@ echo $cluster > cluster-name
 echo "creating cilium ns"
 kubectl create ns cilium || true
 
+echo "deleting terminating namespaces"
+kubectl get ns | grep Terminating | awk '{print $1}' | xargs -n 1 bash -c 'kubectl get ns -o json $1 | tr -d "\n" | sed "s/\"finalizers\": \[[^]]\+\]/\"finalizers\": []/" | kubectl replace --raw /api/v1/namespaces/$1/finalize -f -' -
```
Comment (raybejjani) [inline test/gke/select-cluster.sh]: It is probably also worth it to do this in release-cluster since it's a general purpose cleanup in the end

Comment (raybejjani) [reply test/gke/select-cluster.sh]: It might also be good to explain why it's here and what it fixes (and maybe link to the k8s issue?)

**Review by raybejjani — APPROVED** [review]

File: test/gke/select-cluster.sh (hunk 1 above)
Comment (nebril) [reply test/gke/select-cluster.sh]: good point, done

**Comment by nebril** [comment]
test-gke K8sPolicy.*

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**